hollow/src/main/java/com/netflix/hollow/api/consumer/metrics/ConsumerRefreshMetrics.java

```diff
@@ -0,0 +1,108 @@
+package com.netflix.hollow.api.consumer.metrics;
+
+import com.netflix.hollow.api.consumer.HollowConsumer.Blob.BlobType;
+import java.util.OptionalLong;
+
+public class ConsumerRefreshMetrics {
+
+    private long durationMillis;
+    private boolean isRefreshSuccess;               // true if refresh was successful, false if refresh failed
+    private boolean isInitialLoad;                  // true if initial load, false if subsequent refresh
+    private BlobType overallRefreshType;            // snapshot, delta, or reverse delta
+    private UpdatePlanDetails updatePlanDetails;    // details about the update plan such as no. and types of transitions and no. of successful transitions
+    private long consecutiveFailures;
+    private OptionalLong refreshSuccessAgeMillisOptional; // time elapsed since the previous successful refresh
+    private long refreshEndTimeNano;                // monotonic system time when refresh ended
+
+    public long getDurationMillis() {
+        return durationMillis;
+    }
+    public boolean getIsRefreshSuccess() {
+        return isRefreshSuccess;
+    }
+    public boolean getIsInitialLoad() {
+        return isInitialLoad;
+    }
+    public BlobType getOverallRefreshType() {
+        return overallRefreshType;
+    }
+    public UpdatePlanDetails getUpdatePlanDetails() {
+        return updatePlanDetails;
+    }
+    public long getConsecutiveFailures() {
+        return consecutiveFailures;
+    }
+    public OptionalLong getRefreshSuccessAgeMillisOptional() {
+        return refreshSuccessAgeMillisOptional;
+    }
+    public long getRefreshEndTimeNano() {
+        return refreshEndTimeNano;
+    }
+
+    private ConsumerRefreshMetrics(long durationMillis, boolean isRefreshSuccess, boolean isInitialLoad, BlobType overallRefreshType,
```

Comment:
One pattern I've seen used to great effect is to have the constructor take in the Builder and set fields based on the builder fields.

Reply:
Fits well here, thanks.